sklearn/naive_bayes.py

```diff
@@ -949,3 +988,258 @@ def _joint_log_likelihood(self, X):
+            List of all the classes that can possibly appear in the y vector.
+
+            Must be provided at the first call to partial_fit, can be omitted
+            in subsequent calls.
+
+        Returns
+        -------
+        self : object
+        """
+        return super(CategoricalNB, self).partial_fit(X, y, classes,
+                                                      sample_weight=None)
+
+    def _check_input(self, X, Y=None):
+        if Y is None:
+            if CategoricalNB.old_numpy_:
+                X = check_array(X, accept_sparse=False)
+                if not np.isfinite(X).all():
+                    raise ValueError("Input contains NaN or infinity.")
+                return np.asarray(X, dtype=np.int64)
+            else:
+                return check_array(X, accept_sparse=False)
+        else:
+            if CategoricalNB.old_numpy_:
+                X, Y = check_X_y(X, Y, accept_sparse=False)
+                if not np.isfinite(X).all():
+                    raise ValueError("Input contains NaN or infinity.")
+                X = np.asarray(X, dtype=np.int64)
+                return X, Y
+            else:
+                return check_X_y(X, Y, accept_sparse=False)
+
+    def _init_counters(self, n_effective_classes, n_features):
+        self.class_count_ = np.zeros(n_effective_classes, dtype=np.float64)
+        self.category_count_ = {i: np.zeros((self.class_count_.shape[0], 0))
+                                for i in range(n_features)}
+        self.feature_cat_index_mapping_ = {i: dict() for i in
+                                           range(n_features)}
+
+    def _count(self, X, Y):
+        if np.any((X.data if issparse(X) else X) < 0):
+            raise ValueError("Input X must be non-negative")
+        self.class_count_ += Y.sum(axis=0)
+        for i in range(self.n_features_):
+            X_feature = X[:, i]
+            categories = np.unique(X_feature)
+            self._update_mapping(self.feature_cat_index_mapping_[i],
+                                 categories)
+            # update category_count_dimensions in case partial_fit is used
+            self.category_count_[i] = self._update_category_count_dimensions(
+                self.category_count_[i], self.feature_cat_index_mapping_[i])
+            self._update_category_count(X_feature, Y,
+                                        self.feature_cat_index_mapping_[i],
+                                        self.category_count_[i])
+
+    def _count_categories(self, X_feature):
+        if CategoricalNB.old_numpy_:
+            bins = np.bincount(X_feature)
+            non_zero_bins = np.nonzero(bins)[0]
+            return non_zero_bins, bins[non_zero_bins]
+        else:
+            return np.unique(X_feature, return_counts=True)
+
+    def _update_mapping(self, cat_mapping, categories):
+        for category in np.nditer(categories):
+            category = float(category)
```

Comment:
why a float?

Reply:
Because numpy arrays, even when they contain only one element, as is the case here, are not hashable and can therefore not function as a dictionary key.

Reply:
Ah, that was not true. The problem was with nditer, I forgot to remove it at one place. Now it works.